hack/generated/controllers/generic_controller.go

```diff
@@ -0,0 +1,749 @@
+//}
+
+func (gr *GenericReconciler) reconcileDelete(
+	ctx context.Context,
+	log logr.Logger,
+	metaObj genruntime.MetaObject,
+	f func(context.Context, logr.Logger, genruntime.ArmResource, genruntime.MetaObject) (ctrl.Result, error)) (ctrl.Result, error) {
+
+	log.Info("reconcile delete start")
+
+	_, armResourceSpec, err := ResourceSpecToArmResourceSpec(gr, metaObj)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "couldn't convert to armResourceSpec")
+	}
+	// TODO: Not setting status here
+	resource := genruntime.NewArmResource(armResourceSpec, nil, getResourceId(metaObj))
+
+	return f(ctx, log, resource, metaObj)
+}
+
+// startDeleteOfResource will begin the delete of a resource by telling Azure to start deleting it. The resource will be
+// marked with the provisioning state of "Deleting".
+func (gr *GenericReconciler) startDeleteOfResource(
+	ctx context.Context,
+	log logr.Logger,
+	resource genruntime.ArmResource,
+	metaObj genruntime.MetaObject) (ctrl.Result, error) {
+
+	if err := patcher(ctx, gr.Client, metaObj, func(ctx context.Context, mutMetaObject genruntime.MetaObject) error {
+		if resource.GetId() != "" {
+			emptyStatus, err := NewEmptyArmResourceStatus(metaObj)
+			if err != nil {
+				return errors.Wrapf(err, "failed trying to create empty status for %s", resource.GetId())
+			}
+
+			err = gr.ARMClient.BeginDeleteResource(ctx, resource.GetId(), resource.Spec().GetApiVersion(), emptyStatus)
+			if err != nil {
+				return errors.Wrapf(err, "failed trying to delete resource %s", resource.Spec().GetType())
+			}
+
+			annotations := make(map[string]string)
+			annotations[ResourceStateAnnotation] = string(armclient.DeletingProvisioningState)
+			addAnnotations(metaObj, annotations)
+		} else {
+			controllerutil.RemoveFinalizer(mutMetaObject, GenericControllerFinalizer)
+		}
+
+		// TODO: Not exactly sure what this was doing
+		//if err := gr.Converter.FromResource(resource, mutMetaObject); err != nil {
+		//	return errors.Wrapf(err, "error gr.Converter.FromResource")
+		//}
+
+		return nil
+	}); err != nil && !apierrors.IsNotFound(err) {
+		return ctrl.Result{}, errors.Wrap(err, "failed to patch after starting delete")
+	}
+
+	// delete has started, check back to seen when the finalizer can be removed
+	return ctrl.Result{
+		RequeueAfter: 5 * time.Second,
+	}, nil
+}
+
+// updateFromNonTerminalDeleteState will call Azure to check if the resource still exists. If so, it will requeue, else,
+// the finalizer will be removed.
+func (gr *GenericReconciler) updateFromNonTerminalDeleteState(
+	ctx context.Context,
+	log logr.Logger,
+	resource genruntime.ArmResource,
+	metaObj genruntime.MetaObject) (ctrl.Result, error) {
+
+	// already deleting, just check to see if it still exists and if it's gone, remove finalizer
+	found, err := gr.ARMClient.HeadResource(ctx, resource.GetId(), resource.Spec().GetApiVersion())
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err, "failed to head resource")
+	}
```

Comment:
Wouldn't a missing (ie deleted) resource result in an error?

Reply:
No, the impl of `ARMClient.HeadResource` turns `IsNotFound(err)` into `false`, so the errors returned here are other things (that are actually errors)